turbopack/crates/turbopack-cli-utils/src/issue.rs

```diff
@@ -170,12 +171,105 @@ pub fn format_issue(
             writeln!(styled_issue, "{path}").unwrap();
         }
     }
+    let traces = &*plain_issue.import_traces;
+    if !traces.is_empty() {
+        /// Returns the layer used by all items in the trace if it is unique.
+        /// Returns None, if there are multiple different layers (or no layers)
+        fn get_layer(items: &[PlainTraceItem]) -> Option<RcStr> {
+            let (unique, value) = items
+                .iter()
+                .map(|i| (true, &i.layer))
+                .reduce(|(unique, prev), (_, next)| {
+                    if unique && prev == next {
+                        (true, prev)
+                    } else {
+                        (false, prev)
+                    }
+                })
```

Comment:
Minor nit: I usually find once you're using `reduce` that it would be easier to just write a for loop.

Reply:
indeed.  loop is much simpler.